### PR TITLE
Correctly handle missing dependency

### PR DESCRIPTION
**Pull request by JaxkDev**
Aimed directly at #207 where trying to view submit form for a plugin with a dependency not on poggit would result in the standard 500 page with no actual error being thrown.
Changed this behaviour to take advantage of the built in 404 Not found in SubmitFormAjax.php to tell user that their dependency cannot be found.

Commit used to test release: https://github.com/JaxkDev/PoggitTestPlugin/commit/f6a0b4b469acb014cdcc457a35d7154dcb4bc137

(Any plugin can be tested where depend[] includes a non-existing plugin name.)

(ignore dark theme)

Before:
image

After:
image

### CODE REVIEW

**Comment by SOF3** [comment]
Shouldn't `errorBadRequest` trigger a 400 instead of 500 tho?

**Comment by JaxkDev** [comment]
It does however once it throws that 400 it’s classed as an error and it then reaches this:
https://github.com/poggit/poggit/blob/0c9977b90b573c48c7a75e497705e7b4850e9f84/js/newSubmit.js#L94

**Comment by SOF3** [comment]
seems this needs a fix.

**Comment by JaxkDev** [comment]
I thought this was just a case of calling the wrong function as I see it handles some specific things to submit (https://github.com/poggit/poggit/blob/0c9977b90b573c48c7a75e497705e7b4850e9f84/js/newSubmit.js#L31)

**Comment by SOF3** [comment]
Closed since this is fixing the error at the wrong side.

**Comment by JaxkDev** [comment]
Looking back i cant remember what you said regarding above

fixing error at wrong end?

Looking at it, it could just be a typo `exitBadRequest` instead of `errorBadRequest`

**Comment by JaxkDev** [comment]
errorBadRequest (and others) cause a 500 as they are errors
exitBadRequest (and others) return a success but with error data for client to handle (400)
